### ISSUE TITLE
validation error missing message when using "additionalProperties" and "required" in schema

### ISSUE THREAD

**Issue by Pierrocana** [josdejong/jsoneditor#544]
Hello,

In the the validation schema example: https://github.com/josdejong/jsoneditor/blob/master/examples/07_json_schema_validation.html

If we add to the schema a required and a additionalProperties to the schema like this:

```javascript
var schema = {
    "title": "Example Schema",
    "type": "object",
    "properties": {
      "firstName": {
        "type": ["string", "null"]
      },
      "lastName": {
        "type": "string"
      },
      "gender": {
        "enum": ["male", "female"]
      },
      "age": {
        "description": "Age in years",
        "type": "integer",
        "minimum": 0
      },
      "job": {
        "$ref": "job"
      }
    },
    "additionalProperties": false,
    "required": ["firstName", "lastName"]
  };
```

And then we change a property name of the first level object by something not defined in the schema:

![foo_property](https://user-images.githubusercontent.com/6430904/41727689-3c35df98-7575-11e8-86de-5b8b9718de6c.png)

The editor does not display the error "should have required property" and the error "should NOT have additional property:". The errors are present in editor.validateSchema.errors but the messages are not displayed.

The error messages are missing only when we use both "additionalProperties" and "required" constraints.

**Comment by josdejong** [comment]
Thanks for reporting! Turns out there was a bug which caused _no_ errors to be displayed when the root node contains an even number of errors, like two in your example.

Fixed via https://github.com/josdejong/jsoneditor/commit/fbf20b50311f7190f5a9c63a87cdef282e48d0ee (not yet released)

**Comment by Pierrocana** [comment]
Ok, thank you.

**Comment by josdejong** [comment]
Should be fixed now in v5.18.0
